pageserver/src/disk_usage_eviction_task.rs

```diff
@@ -263,14 +274,13 @@ struct LayerCount {
 
 #[allow(clippy::needless_late_init)]
 pub async fn disk_usage_eviction_task_iteration_impl<U: Usage>(
+    state: &DiskUsageEvictionState,
     storage: &GenericRemoteStorage,
     usage_pre: U,
     cancel: &CancellationToken,
 ) -> anyhow::Result<IterationOutcome<U>> {
-    static MUTEX: once_cell::sync::Lazy<tokio::sync::Mutex<()>> =
-        once_cell::sync::Lazy::new(|| tokio::sync::Mutex::new(()));
-
-    let _g = MUTEX
+    let _g = state
+        .mutex
```

Comment:
Yeah this could be whatever "simpler", but Semaphore is already an atomic counter for this purpose so it's not really that bad. Originally we waited.

Reply:
This is fine for me.